backends/cuda-gen/ceed-cuda-gen-qfunction.c

```diff
@@ -126,15 +60,16 @@ int CeedQFunctionCreate_Cuda_gen(CeedQFunction qf) {
   ierr = CeedCalloc(1, &data); CeedChkBackend(ierr);
   ierr = CeedQFunctionSetData(qf, data); CeedChkBackend(ierr);
 
-  char *source;
-  ierr = CeedQFunctionGetSourcePath(qf, &source); CeedChkBackend(ierr);
-  const char *funname = strrchr(source, ':') + 1;
-  data->qFunctionName = (char *)funname;
-  const int filenamelen = funname - source;
-  char filename[filenamelen];
-  memcpy(filename, source, filenamelen - 1);
-  filename[filenamelen - 1] = '\0';
-  ierr = loadCudaFunction(qf, filename); CeedChkBackend(ierr);
+  // Read QFunction source
+  ierr = CeedQFunctionGetKernelName(qf, &data->qFunctionName);
+  CeedChkBackend(ierr);
+  ierr = CeedQFunctionLoadSourceToBuffer(qf, &data->qFunctionSource);
+  CeedChkBackend(ierr);
+  if (!data->qFunctionSource)
+    // LCOV_EXCL_START
+    return CeedError(ceed, CEED_ERROR_UNSUPPORTED,
+                     "/gpu/cuda/gen backend requires QFunction source code file");
+  // LCOV_EXCL_STOP
```

Comment:
Why is this code outside `CeedQFunctionLoadSourceToBuffer`?

Reply:
As I said in the other PR, I would prefer details of this kind to be inside a function.

Reply:
As I said in the other PR, it is not a problem if no QFunction source file exists for `/gpu/cuda/ref`, `/gpu/cuda/shared`, `/gpu/hip/ref`, and `/gpu/hip/shared`. Only the `/gpu/cuda/gen` and `/gpu/hip/gen` backends cannot accept a user provided `CUfunction`/`hipFunction_t` in place of a source file.

This check is in the only two places where it needs to be.